fusesoc/config.py

```diff
@@ -149,13 +162,14 @@ def add_library(self, name, library):
 
         self.libraries[name] = library
 
-        if not os.path.isdir(library['location']):
-            logger.info("Cloning library {}".format(name))
-            git_args = ['clone', library['sync-uri'], library['location']]
-            try:
-                Launcher('git', git_args).run()
-            except RuntimeError as e:
-                logger.error("`library add` failed: " + str(e))
+        try:
+            provider_module = importlib.import_module(
+                'fusesoc.provider.%s' % library['sync-type'])
+        except ImportError as e:
+            logger.error("Invalid sync-type '{}'".format(library['sync-type']))
+            exit(1)
+
+        provider_module.PROVIDER_CLASS.init_library(library)
```

Comment:
Had forgotten about this PROVIDER_CLASS thing, and I think it's a bit unnecessary. No big deal, but we could do something like the _import function in main.py

Reply:
Done as a second commit

Reply:
Awesome. I was thinking of suggesting to move it to utils.py and clean up the other references to PROVIDER_CLASS, but you had already done exactly that :)